File: Python/8.py
import random
def get_random_item():
 items=["sword","shield","potion","armor","ring"]
 return random.choice(items)
def encounter_monster():
 monsters=["goblin","troll","dragon","orc"]
 return random.choice(monsters)
def play_turn(player,health):
 action=input("Choose action (fight/run):")
 if action=="fight":
  monster=encounter_monster()
  print("You encounter a",monster)
  damage=random.randint(5,20)
  health-=damage
  print("You take",damage,"damage")
 elif action=="run":
  print("You run away")
 else:
  print("Invalid action")
 return health
def game_loop():
 player=input("Enter your name:")
 health=100
 inventory=[]
 turns=0
 while health>0 and turns<10:
  print("Turn",turns+1,"Health:",health)
  choice=input("Choose (explore/rest):")
  if choice=="explore":
   item=get_random_item()
   print("You found a",item)
   inventory.append(item)
   health=play_turn(player,health)
  elif choice=="rest":
   print("You rest and recover 10 health")
   health+=10
  else:
   print("Invalid choice")
  turns+=1
  for i in range(2):
   print("Extra event",i)
 print("Game over")
 print("Final Health:",health)
 print("Inventory:",inventory)
 return health,inventory
def bonus_round():
 score=0
 for i in range(5):
  num=random.randint(1,10)
  guess=int(input("Guess a number between 1 and 10:"))
  if guess==num:
   print("Correct")
   score+=10
  else:
   print("Wrong, number was",num)
 return score
print("Welcome to Fantasy Adventure")
final_health,inventory=game_loop()
print("Bonus round begins")
score=bonus_round()
print("Your final score is",score)
for x in range(3):
 for y in range(3):
  print("Nested loop",x,y)
print("Thank you for playing")
for a in range(2):
 print("Extra loop",a)
for b in range(2):
 for c in range(2):
  print("Double nested",b,c)
print("Game simulation complete")
for d in range(3):
 print("Loop",d)
score_total=score+final_health
print("Total score:",score_total)
for e in range(2):
 for f in range(3):
  print("Another nested",e,f)
print("Goodbye, adventurer")
for g in range(3):
 print("Final loop",g)
for h in range(2):
 print("Extra final",h)
print("Simulation end")
for i in range(2):
 for j in range(2):
  print("Additional nested",i,j)
for k in range(3):
 print("Closing loop",k)
print("Final simulation complete")
print("Adventure over")
for m in range(2):
 print("Extra step",m)
for n in range(2):
 print("Step complete",n)
print("All done")
print("Ending simulation")
print("Thank you")
print("See you next time")
print("Goodbye")

File: JavaScript/8.js
function getRandomItem(){
  var items=["sword","shield","potion","armor","ring"];
  return items[Math.floor(Math.random()*items.length)];
}
function encounterMonster(){
  var monsters=["goblin","troll","dragon","orc"];
  return monsters[Math.floor(Math.random()*monsters.length)];
}
function playTurn(player, health){
  var action = prompt("Choose action (fight/run):");
  if(action==="fight"){
    var monster = encounterMonster();
    console.log("You encounter a "+monster);
    var damage = Math.floor(Math.random()*16)+5;
    health -= damage;
    console.log("You take "+damage+" damage");
  } else if(action==="run"){
    console.log("You run away");
  } else {
    console.log("Invalid action");
  }
  return health;
}
function gameLoop(){
  var player = prompt("Enter your name:");
  var health = 100;
  var inventory = [];
  var turns = 0;
  while(health>0 && turns<10){
    console.log("Turn "+(turns+1)+" Health: "+health);
    var choice = prompt("Choose (explore/rest):");
    if(choice==="explore"){
      var item = getRandomItem();
      console.log("You found a "+item);
      inventory.push(item);
      health = playTurn(player, health);
    } else if(choice==="rest"){
      console.log("You rest and recover 10 health");
      health += 10;
    } else {
      console.log("Invalid choice");
    }
    turns++;
    for(var i=0;i<2;i++){
      console.log("Extra event "+i);
    }
  }
  console.log("Game over");
  console.log("Final Health: "+health);
  console.log("Inventory: "+inventory);
  return health;
}
function bonusRound(){
  var score = 0;
  for(var i=0;i<5;i++){
    var num = Math.floor(Math.random()*10)+1;
    var guess = parseInt(prompt("Guess a number between 1 and 10:"));
    if(guess===num){
      console.log("Correct");
      score += 10;
    } else {
      console.log("Wrong, number was "+num);
    }
  }
  return score;
}
console.log("Welcome to Fantasy Adventure");
var finalHealth = gameLoop();
console.log("Bonus round begins");
var score = bonusRound();
console.log("Your final score is "+score);
for(var x=0;x<3;x++){
  for(var y=0;y<3;y++){
    console.log("Nested loop "+x+","+y);
  }
}
console.log("Thank you for playing");
for(var a=0;a<2;a++){
  console.log("Extra loop "+a);
}
for(var b=0;b<2;b++){
  for(var c=0;c<2;c++){
    console.log("Double nested "+b+","+c);
  }
}
console.log("Game simulation complete");
for(var p=0;p<3;p++){
  console.log("Loop "+p);
}
var totalScore = score + finalHealth;
console.log("Total score: "+totalScore);
for(var i=0;i<2;i++){
  console.log("Additional loop "+i);
}
console.log("Adventure over");
for(var j=0;j<2;j++){
  console.log("Final loop "+j);
}
console.log("Simulation ends now");
for(var k=0;k<2;k++){
  for(var l=0;l<2;l++){
    console.log("Nested final "+k+","+l);
  }
}
console.log("Thank you");
console.log("Goodbye");
